proof.go

```diff
@@ -47,6 +48,39 @@ type Proof struct {
 	isMaxNamespaceIDIgnored bool
 }
 
+type jsonProof struct {
+	Start                   int      `json:"start"`
+	End                     int      `json:"end"`
+	Nodes                   [][]byte `json:"nodes"`
+	LeafHash                []byte   `json:"leaf_hash"`
```

Comment:
perhaps stupid question, but why not add the json tags directly to the proof struct?

Reply:
Unfortunately json tags require exported fields. Double checked and get the error:

```
Struct field 'isMaxNamespaceIDIgnored' has 'json' tag but is not exported
```

Reply:
makes sense!